### PR TITLE
[onert/python] Introduce `eval_step`

### PR DESCRIPTION
**Pull request by ragmani**
This commit introduce `eval_step` and common batch logic into `_batch_step`.
- Introduce private helper `_batch_step(inputs, expecteds, update_weights)` to handle binding inputs, running session, collecting losses/metrics, and timing
- Update `train_step` to call `_batch_step(..., update_weights=True)` and expose `"train_time"`
- Add `eval_step` for forward‑only inference, calling `_batch_step(..., update_weights=False)` and exposing `"eval_time"`

For https://github.com/Samsung/ONE/issues/15172#issuecomment-2814288794

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/api/python/package/experimental/train/session.py [hunk 1]
```diff
@@ -246,14 +246,17 @@ def _check_batch_size(self,
                     f"{data_type} batch size mismatch at index {idx}: "
                     f"shape[0] = {arr.shape[0]} vs batch size = {batch_size}")
 
-    def train_step(self, inputs: List[np.ndarray],
-                   expecteds: List[np.ndarray]) -> Dict[str, Any]:
+    def train_step(self,
+                   inputs: List[np.ndarray],
+                   expecteds: List[np.ndarray],
+                   train: bool = True) -> Dict[str, Any]:
```
Comment (seockho-kim) [inline runtime/onert/api/python/package/experimental/train/session.py:252]: Just a question, 
if `train` is `false`, what is the operation of this function (`train_step`)?

Comment (seockho-kim) [inline runtime/onert/api/python/package/experimental/train/session.py:252]: Just a question, 
if `train` is `false`, what is the operation of this function (`train_step`)?

Comment (ragmani) [reply runtime/onert/api/python/package/experimental/train/session.py:252]: It acts like an “eval step", we can get losses and metrics but the model’s weights stay unchanged :)

Comment (seockho-kim) [reply runtime/onert/api/python/package/experimental/train/session.py:252]: It's a little weird to me, because `train` function acts like a eval() function in some case.
But I'm not sure this is normal in these kind of codes.

Comment (ragmani) [reply runtime/onert/api/python/package/experimental/train/session.py:252]: It might be a bit confusing to have the `train_step()` function behave differently based on the `train` flag. In standard practice, it's better to separate the concerns by defining two distinct functions:

- `train_step()`: This is for the training phase, where weights are updated.

- `eval_step()`: This function is for inference or validation, where no weight updates happen and you likely want to disable gradients (using `torch.no_grad()`).

This mirrors how frameworks like PyTorch and PyTorch Lightning handle the training and evaluation phases. In PyTorch, `model.train()` and `model.eval()` toggle the mode for training and inference, respectively, and it's common to have separate functions for each phase. Instead of toggling behavior with a flag inside `train_step()`, it's cleaner to have separate functions to keep things clear and consistent.

Comment (ragmani) [reply runtime/onert/api/python/package/experimental/train/session.py]: I introduced `eval_step()` instead of adding the `train` flag.
